Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -55,6 +58,15 @@ public struct AbstractInterpreter {
         executeOuterEffects(instr)
 
         switch instr.op {
+        case is BeginObjectDefinition,
+            is BeginGetter,
```

Comment:
Is this correct? I think only the `BeginObjectDefinition` creates a child state, the others create sibling states? Not 100% sure though.
A child state basically builds onto the parent state, so e.g. inside a `BeginIf`, the changes from the previous code are still visible. On the other hand, a sibling state discards the changes from its other siblings. E.g. inside the `else` branch of an `if`, the changes from the `if` body are no longer valid

Reply:
Thanks for the explanation, I think here (under `execute`) we don't need to handle state changes for `BeginObjectDef` and `EndObjectDef`. Although we are starting a new block, some of the non-block-opening instructions in the block (e.g. `CreateProperty`) don't affect the parent state or the sibling states. We only need to care about getter and setter blocks since they affect the parent state.

Note: `BeginObjDef` and `EndObjDef` are being analysed in `executeOuterEffects` so that we can track properties that have been added to the object being created.